### PR TITLE
Adding definitions for fastGet, fastPut in ssh2-sftp-client

### PR DESCRIPTION
**Pull request by kartik2406**
ssh2-sftp-client 2.3.0 adds support for 'ssh-streams' fastGet and fastPut method, however the typings file does not contain this information
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/mscdex/ssh2-streams/blob/master/SFTPStream.md>
- [X] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/ssh2-sftp-client/index.d.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@
...
-
+import * as ssh2Stream from 'ssh2-streams';
 export = sftp;
 
 declare class sftp {
   connect(options: ssh2.ConnectConfig): Promise<void>;
   list(remoteFilePath: string): Promise<sftp.FileInfo[]>;
   get(remoteFilePath: string, useCompression?: boolean, encoding?: string | null): Promise<NodeJS.ReadableStream>;
-  fastGet(remoteFilePath: string, localPath: string, options?: any): Promise<any>;
+  fastGet(remoteFilePath: string, localPath: string, options?: ssh2Stream.TransferOptions): Promise<any>;
```
Comment (ascariandrea) [inline types/ssh2-sftp-client/index.d.ts]: Does `ssh2-streams` have also a type for the returned value of the `Promise`?

Comment (igrayson) [reply types/ssh2-sftp-client/index.d.ts]: From [the implementation](https://github.com/jyu213/ssh2-sftp-client/pull/64/files), it looks like we should type this as a `Promise<string>`.

Comment (ascariandrea) [reply types/ssh2-sftp-client/index.d.ts]: Yep, can you change it?

Comment (kartik2406) [reply types/ssh2-sftp-client/index.d.ts]: Great catch !! Made the changes :smile:

**Review by ascariandrea — APPROVED** [review]

File: types/ssh2-sftp-client/ssh2-sftp-client-tests.ts [hunk 2]
```diff
@@ -15,10 +15,14 @@ client.get('/remote/path').then(stream => stream.read(0));
 client.get('/remote/path', true, 'utf8').then(stream => stream.read(0));
 client.get('/remote/path', true, null).then(stream => stream.read(0));
 
+client.fastGet('/remote/path', 'local/path').then(() => null);
+
 client.put('/local/path', '/remote/path').then(() => null);
 client.put(new Buffer('content'), '/remote/path').then(() => null);
 client.put(fs.createReadStream('Hello World'), '/remote/path').then(() => null);
 
+client.fastPut('/remote/path', 'local/path').then(() => null);
```
Comment (igrayson) [inline types/ssh2-sftp-client/ssh2-sftp-client-tests.ts]: This should say `fastPut('/local/path', '/remote/path' ...`

**Review by igrayson — APPROVED** [review]

**Comment by igrayson** [comment]
Just a nit.

**Pull request merged**